api/v1/endpoint_types.go

```diff
@@ -141,15 +155,235 @@ type VisibilityConfig struct {
 	// +required
 	Enable bool `json:"enable"`
 	// +optional
-	APISettings *EndpointAPISettingsSpec `json:"apiSettings,omitempty"`
+	Policies []Policy `json:"policies,omitempty"`
+}
+
+//==============================================================================
+// API-M Policy Configuration
+//==============================================================================
+
+// Policy defines an API management policy for an endpoint
+type Policy struct {
+	// +required
+	Name string `json:"name"`
+	// +required
+	Type string `json:"type"`
+	// +optional
+	// enabled if not specified
+	Enabled *bool `json:"enabled,omitempty"`
+	// +required
+	*PolicySpec `json:"policySpec"`
+}
+
+// PolicySpec defines the configuration for different types of policies
+type PolicySpec struct {
+	// +optional
+	APIKeyAuth *APIKeyAuthPolicySpec `json:"apiKeyAuth,omitempty"`
+	// +optional
+	BasicAuth *BasicAuthPolicySpec `json:"basicAuth,omitempty"`
+	// +optional
+	OAuth2 *OAuth2PolicySpec `json:"oauth2,omitempty"`
+	// +optional
+	RateLimit *RateLimitPolicySpec `json:"rateLimit,omitempty"`
+	// +optional
+	CORS *CORSPolicySpec `json:"cors,omitempty"`
+	// +optional
+	MediationPolicies *[]MediationPolicy `json:"mediationPolicies,omitempty"`
```

Comment:
Shall we add some validations?
https://book.kubebuilder.io/reference/markers/crd-validation

Reply:
as we discussed offline, lets skip this validation for the moment